PowerEditor/src/WinControls/ProjectPanel/ProjectPanel.cpp

```diff
@@ -772,9 +1261,25 @@ POINT ProjectPanel::getMenuDisplyPoint(int iButton)
 	return p;
 }
 
+
+const std::vector<generic_string>* ProjectPanel::getFilters(HTREEITEM hItem)
+{
+	TVITEM tvItem;
+	tvItem.mask = TVIF_PARAM;
+	tvItem.hItem = hItem;
+	::SendMessage(_treeView.getHSelf(), TVM_GETITEM, 0, reinterpret_cast<LPARAM>(&tvItem));
+	ProjectPanelData& data = *reinterpret_cast<ProjectPanelData*>(tvItem.lParam);

```

Comment:
You ***absolutely*** should check to see if `::SendMessage` returned `FALSE` before using `tvItem.lParam`.